### ISSUE TITLE
replace the folium module with geopandas.GeoDataFrame.explore

### ISSUE THREAD

**Issue by gboeing** [gboeing/osmnx#935]
Shapely and GeoPandas have recently had some updates that raise a couple broad questions for OSMnx:

1. Is OSMnx's `folium` module useful given `geopandas.GeoDataFrame.explore`'s functionality?

I think we should deprecate the `folium` module in favor of using `geopandas.GeoDataFrame.explore` directly.

2. Is OSMnx's coordinate precision functionality useful given `shapely.set_precision`'s functionality?

No action for this one. I think the answer is yes. In particular, we use this to minimize file size when serializing graphs that otherwise would have attributes with arbitrarily fine precision. It may be worth considering moving precision from function-level arguments to the settings module at some point.

3. Should the `plot_orientation` function be moved from the bearing module to the plot module.

For consistency yes. See #956 for this work.

**Comment by gboeing** [comment]
Also, `plot_orientation` should probably be moved from the bearing module to the plot module.

**Comment by gboeing** [comment]
> Is OSMnx's coordinate precision functionality useful given shapely.set_precision's functionality?

I think the answer is yes. In particular, we use this to minimize file size when serializing graphs that otherwise would have attributes with arbitrarily fine precision. It may be worth considering moving precision from function-level arguments to the settings module at some point.

**Comment by gboeing** [comment]
Replacing the `folium` module with `geopandas.GeoDataFrame.explore` could look something like this:

```python
import osmnx as ox
G = ox.graph_from_place('Piedmont, CA, USA', network_type='drive')

# explore the edges the current way with OSMnx
ox.folium.plot_graph_folium(G)

# explore the edges directly with geopandas
ox.graph_to_gdfs(G, nodes=False).explore(tiles='cartodbpositron')

# can't explore the nodes with OSMnx currently...
# but you can explore the nodes directly with geopandas
ox.graph_to_gdfs(G, edges=False).explore(tiles='cartodbpositron')

weight = 'length'
path = ox.shortest_path(G, list(G.nodes)[0], list(G.nodes)[-1], weight=weight)

# explore a route the current way with OSMnx
ox.folium.plot_route_folium(G, path)

# explore a route directly with geopandas
uvk = [(u, v, min(G.get_edge_data(u, v).items(), key=lambda x: x[1][weight])[0]) for u, v in zip(path[:-1], path[1:])]
ox.graph_to_gdfs(G, nodes=False).loc[uvk].explore(tiles='cartodbpositron')
```

**Comment by gboeing** [comment]
See work and further discussion in #957.

**Comment by gboeing** [comment]
Closed by #957
